beetsplug/absubmit.py

```diff
@@ -115,29 +130,33 @@ def _get_analysis(self, item):
         # the data into a python object and then remove the file from the
         # system.
         tmp_file, filename = tempfile.mkstemp(suffix='.json')
+        # Close the file, so the extractor can overwrite it.
         try:
-            # Close the file, so the extractor can overwrite it.
-            try:
-                call([self.extractor, util.syspath(item.path), filename])
-            except ABSubmitError as e:
-                self._log.error(
-                    u'Failed to analyse {item} for AcousticBrainz: {error}',
-                    item=item, error=e
-                )
-                return None
-            with open(filename) as tmp_file:
-                analysis = json.loads(tmp_file.read())
-            # Add the hash to the output.
-            analysis['metadata']['version']['essentia_build_sha'] = \
-                self.extractor_sha
-            return analysis
-        finally:
-            try:
-                os.remove(filename)
-            except OSError as e:
-                # errno 2 means file does not exist, just ignore this error.
-                if e.errno != 2:
-                    raise
+            call([self.extractor, util.syspath(item.path), filename])
+        except ABSubmitError as e:
+            self._log.error(
+                u'Failed to analyse {item} for AcousticBrainz: {error}',
+                item=item, error=e
+            )
+            return None
```

Comment:
It looks like the temporary file is not removed if the analysis fails. This is why the whole part was in a try/finally